### ISSUE TITLE
Rook mon failover stuck due to garbage directory

### ISSUE THREAD

**Issue by danderson** [rook/rook#1750]
* Bug Report

**Bug Report**

What happened:

On a test cluster where I rebooted nodes a bunch of times without first draining, I managed to get rook into a wedged state. I think what happened is that I rebooted a node while Rook was in the middle of failing over a mon onto it.

The situation: cluster has 3 nodes, `prod-{01,02,03}`, and one Rook-managed Ceph cluster on it. The operator has 3 mons, one per node, everything's happy.

I reboot `prod-02` without draining k8s (`sudo reboot` over ssh). One of the mons goes down, the operator fails it over to `prod-01`. `prod-01` is now hosting 2 mons, `prod-03` is hosting 1.

`prod-02` comes back online some minutes later. Rook notices, and begins another failover to spread the mons back out.

With terrible timing, I reboot `prod-02` again, again ungracefully. The new mon never comes up, Rook aborts the failover and unwinds cleanly back to the existing state, with 2 mons on `prod-01`.

`prod-02` comes back again, and Rook once again starts a failover. However, this failover never succeeds. The newly created mon pod is exiting (with exit code 0) after a few seconds, and so goes into CrashloopBackoff. Eventually, rook gives up, removes the crashlooping replicaset, and tries again.

The only clue I could find in the logs of the crashlooping mon was:

```
2018-05-28 07:00:07.061504 I | rook-ceph-mon13: 2018-05-28 07:00:07.061418 7f3a1b612700  0 mon.rook-ceph-mon13@3(probing) e27  removed from monmap, suicide.
```

So, this mon had apparently decided that it should not be part of the cluster, and was correctly exiting as a result.

Confused, I ssh'd into `prod-02` and `rm -rf /var/lib/rook/rook-ceph-mon13`. The next time Rook attempted the failover, mon13 came up happily, and the failover was able to complete.

My best guess as to what happened, is that somehow mon13's local state recorded that it was no longer part of the cluster (which it wasn't, after the first aborted failover due to the reboot)... But Rook kept trying to reuse the mon13 identity, and was not cleaning up the local on-disk state from the previous failover attempt, so mon13 was constantly coming up with stale state from a past failover attempt, and stayed wedged in a belief that it should not be running. Once I manually cleaned up the old mon state, mon13 was able to bootstrap and join the quorum.

What you expected to happen:

I expect Rook to be perfect at all times! ;) More seriously, I expect Rook to be resistant to node disappearances at any stage of its lifecycle, as long as the mons still have quorum. In particular, a node loss at a critical phase of a failover should not wedge the operator.

How to reproduce it (minimal and precise):

Unfortunately I don't have a precise recipe, since I cleverly nuked the incriminating state instead of saving a copy... But the above timeline captures what I think happened. If I'm correct, a judicious hard reboot mid-failover should be enough to trigger this wedging.

**Environment**:
* OS (e.g. from /etc/os-release): Debian Buster (testing)
* Kernel (e.g. `uname -a`): 4.16.5
* Cloud provider or hardware configuration: Bare metal
* Rook version (use `rook version` inside of a Rook Pod): v0.7.1
* Kubernetes version (use `kubectl version`): 1.10.3
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm
* Ceph status (use `ceph health` in the [Rook toolbox](https://Rook.io/docs/Rook/master/toolbox.html)): HEALTH_OK (3 mons up, 3 osds up, 3 osds in)

**Comment by bjornryden** [comment]
I've just experienced the same thing. I have before and after contents of the rook-ceph-mon-X directory if anyone is interested.

**Comment by travisn** [comment]
@bjornryden could you also get the operator logs during this time? Thanks!

**Comment by travisn** [comment]
I believe this would be fixed already by #2631 and it was also included in the v0.9.3 release.
